sample/src/main/java/com/metaplex/sample/PhantomLoginFragment.kt

```diff
@@ -0,0 +1,154 @@
+package com.metaplex.sample
+
+import android.content.Context.MODE_PRIVATE
+import android.content.Intent
+import android.content.SharedPreferences
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import androidx.navigation.NavController
+import androidx.navigation.fragment.findNavController
+import com.google.android.material.snackbar.Snackbar
+import com.metaplex.sample.databinding.FragmentPhantomLoginBinding
+import com.solana.networking.Network
+import com.solana.vendor.TweetNaclFast
+import org.bitcoinj.core.Base58
+import java.net.URL
+
+
+class PhantomLoginFragment : Fragment() {
+    private var _binding: FragmentPhantomLoginBinding? = null
+    private val binding get() = _binding!!
```

Comment:
dont use the null assertion operator (!!), this will crash the app. We want to avoid throwing fatal errors when we encounter an issue.

Reply:
@Funkatronics This is the recommended way to use view binding in fragments as mentioned in the docs: https://developer.android.com/topic/libraries/view-binding#fragments
The null assertion will work since it is ALWAYS valid between `onCreateView` and `onDestroyView`.

Reply:
the docs do not advocate using the null assertion, and there is actually no need to store a reference to the binding in this case. In my own fragments, I only ever save a reference to the binding if I actually need to reference it at runtime. In this case, we actually could leave all access tot he binding in onCreateView. This is what I would recommend. 

However, you are correct, what you have here will work okay. we can leave it as is for now and clean it up when we re-arch 👌